utils/api_helpers.py

```diff
@@ -35,27 +61,38 @@ async def analyse_image(image_url: str) -> str:
                 ],
             }
         ],
-        model=VISION_MODEL,
+        model=vision_model,
     )
     return image_completion.choices[0].message.content
 
-
-async def generate_single(prompt: str) -> str:
-    client = AsyncGroq(api_key=config.groq_api_key)
-    chat_completion = await client.chat.completions.create(messages=[{"role": "user", "content": prompt}],
-                                                           model=LANGUAGE_MODEL, max_tokens=400)
+async def generate_single(prompt: str) -> str: # For future me, this is never used as of 2024-12-21 YYYY-MM-DD
+    client, language_model, vision_model = get_client()
+    try:
+        chat_completion = await client.chat.completions.create(messages=[{"role": "user", "content": prompt}],
+                                                            model=language_model, max_tokens=400)
+    except Exception as err:
+        log.error(
+            f"An error occurred while generating single: {err}",
+            exc_info=(type(err), err, err.__traceback__))
+        return "An error occurred. Please try again later, this has been logged."
```

Comment:
Remove this error handling too, it's handled in the global error handler

Reply:
My bad, I thought i removed it earlier.